src/XML/src/FMI3/fmi3_xml_model_structure.c

```diff
@@ -476,14 +545,46 @@ int fmi3_xml_handle_InitialUnknown(fmi3_xml_parser_context_t *context, const cha
     return 0;
 }
 
-// TODO
 int fmi3_xml_handle_EventIndicator(fmi3_xml_parser_context_t *context, const char* data) {
     if(!data) {
         fmi3_xml_model_description_t* md = context->modelDescription;
         fmi3_xml_model_structure_t* ms = md->modelStructure;
 
-        //return fmi3_xml_parse_unknown(context, fmi3_xml_elmID_EventIndicator, &ms->eventIndicators, ms->eventIndicatorDeps);
-        return -1; // TODO
+        // Ignored if Co-simulation & scheduled execution
+        fmi3_fmu_kind_enu_t fmu_kind = fmi3_fmu_kind_unknown;
+        fmu_kind = fmi3_xml_get_fmu_kind(md);
+        // Multiple types can be defined, check for: not ME and (CS or SE)
```

Comment:
Not sure what the best way is here, since one can define multiple types.

Reply:
This looks good to me